src/frontend/src/tests/components/send/SendInputAmount.spec.ts

```diff
@@ -0,0 +1,47 @@
+import SendInputAmount from '$lib/components/send/SendInputAmount.svelte';
+import en from '$tests/mocks/i18n.mock';
+import { fireEvent, render } from '@testing-library/svelte';
+
+describe('SendInputAmount', () => {
+	const amount = 10.25;
+
+	const props = {
+		amount: amount,
+		calculateMax: () => amount * 2
+	};
+
+	it('should render an input', () => {
+		const { container } = render(SendInputAmount, { props });
+
+		const input: HTMLInputElement | null = container.querySelector('input');
+		expect(input?.value).toBe(props.amount.toString());
+	});
+
+	it('should not render a max button if calculateMax is not provided', () => {
+		const { calculateMax: _, ...newProps } = props;
+		const { container } = render(SendInputAmount, { props: newProps });
+
+		const button: HTMLButtonElement | null = container.querySelector('button');
+		expect(button).toBeNull();
+	});
+
+	it('should render a max button if calculateMax is provided', () => {
+		const { container, getByText } = render(SendInputAmount, { props });
+
+		const button: HTMLButtonElement | null = container.querySelector('button');
+		expect(button).not.toBeNull();
+		expect(getByText(en.core.text.max)).toBeInTheDocument();
+	});
+
+	it('should trigger max value', async () => {
+		const { container } = render(SendInputAmount, { props });
+
+		const button: HTMLButtonElement = container.querySelector('button') as HTMLButtonElement;
+		if (!button) throw new Error('Button not found');
```

Comment:
if interesting, we have a utils called `assertNonNullish` which assert that something is not null or undefined, throw and error and inform typescript about the fact that the element is valid

```
assertNonNullish(button);
```

Reply:
Solved